extensions/ql-vscode/src/vscode-tests/cli-integration/remote-queries/run-remote-query.test.ts

```diff
@@ -58,6 +60,9 @@ describe('Remote queries', function() {
     // always run in the vscode-codeql repo
     await setRemoteControllerRepo('github/vscode-codeql');
     await setRemoteRepositoryLists({ 'vscode-codeql': ['github/vscode-codeql'] });
+
+    // Consider live results disabled unless specifically enabled in a test
+    await setVariantAnalysisLiveResultsEnabled(false);
```

Comment:
We are now setting this value before each test, but that means that the value after this test file has run is that the value might be set to true depending on the order of tests. Since other test files will execute in the same VSCode instance, that might result in some different behavior for those tests as well.

I think it would be better to move this to the `afterEach` hook to reset the value after every test; that will ensure that the value will be `false` for certain after the test file has completed.

Reply:
Thanks for bringing this up. So values like this aren't reset between tests? I wasn't sure so I added this before each test, but was hoping to clarify before merging the PR.

Unless it's too expensive (which I doubt) should we do it before and after, so we don't rely on other tests leaving a consistent state? I note we don't go to those lengths for the other config settings, but that's a different issue and doesn't mean we can't do something else.

What I'd like even more is a `resetConfig` method that would set all of the config to a consistent state. Then every test could call that in its `beforeEach` method and not have to worry about any leftover config changes. If you think that's a good idea I can open an issue to look at it.

Reply:
I don't think the config is reset between each test, the only integration between VSCode and the test runner is the script in [`extensions/ql-vscode/src/vscode-tests/index-template.ts`](https://github.com/github/vscode-codeql/blob/8c7c197b2200a7547feafb052644fd7373239f04/extensions/ql-vscode/src/vscode-tests/index-template.ts) which doesn't seem to reset the config.

I think it would indeed be a good idea to have a `resetConfig` function which will reset all config values. We would then be able to run this before every test and have a consistent state without needing to reset the state before/after tests.